sputnikdao-factory2/src/lib.rs

```diff
@@ -3,15 +3,31 @@ mod factory_manager;
 use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
 use near_sdk::collections::UnorderedSet;
 use near_sdk::json_types::{Base58CryptoHash, Base64VecU8, U128};
+use near_sdk::serde::{Deserialize, Serialize};
 use near_sdk::serde_json::{self, json};
 use near_sdk::{env, near_bindgen, AccountId, CryptoHash, PanicOnDefault, Promise};
 
 use factory_manager::FactoryManager;
 
+// The keys used for writing data to storage via `env::storage_write`.
+const DEFAULT_CODE_HASH_KEY: &[u8; 4] = b"CODE";
+const FACTORY_OWNER_KEY: &[u8; 5] = b"OWNER";
+const CODE_METADATA_KEY: &[u8; 8] = b"METADATA";
+
+// The values used for writing data to storage via `env::storage_write`.
 const DAO_CONTRACT_CODE: &[u8] = include_bytes!("../../sputnikdao2/res/sputnikdao2.wasm");
+const DAO_CONTRACT_VERSION: &str = "V3";
+const DAO_CONTRACT_NO_DATA: &str = "unavailable";
 
-const LATEST_CODE_HASH_KEY: &[u8; 4] = b"CODE";
-const OWNER_KEY: &[u8; 5] = b"OWNER";
+#[derive(BorshSerialize, BorshDeserialize, Serialize, Deserialize)]
+#[cfg_attr(not(target_arch = "wasm32"), derive(Clone, Debug))]
+#[serde(crate = "near_sdk::serde")]
+pub struct DaoContractMetadata {
+    pub code_hash: Base58CryptoHash,
+    pub version: String,
+    pub commit_id: String,
+    pub readme: String,
```

Comment:
I don't think this is a big deal, but if we have the commit ID, you can see the README.md (or perhaps it's better if we created a markdown CHANGELOG.md like near-sdk-rs has with anchors to version changes: https://github.com/near/near-sdk-rs/blob/master/CHANGELOG.md#400-pre5-12-23-2021)

https://user-images.githubusercontent.com/1042667/151470608-626d6985-c0e1-41b1-96e8-4c469efc5a99.mp4

Reply:
So we may have enough info just given the commit it. But this is not strongly held as it's not that costly to have one extra field for `readme`

Reply:
`readme` was meant to be more like a changelog, than the current readme from the repo. But you're right, the name should reflect that. I've changed the name and also made the parameter optional. Thanks!